CMakeLists.txt

```diff
@@ -17,40 +17,57 @@ if (LEGACY_BUILD)
     project("aws-cpp-sdk-all" VERSION "${PROJECT_VERSION}" LANGUAGES CXX)
     include(legacy_main)
 else ()
-    message(STATUS "Building with 1.10 CMake scripts.")
+    message(STATUS "Building with new CMake scripts.")
     string(CONCAT DESCRIPTION_STRING "The AWS SDK for C++ provides a modern C++ (standard version C++11 or later) "
             "interface for Amazon Web Services (AWS).")
 
-    if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
-        message(STATUS "Setting build type to 'RelWithDebInfo' as none was specified.")
-        set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE
-                STRING "Choose the type of build." FORCE)
-        # cmake-gui helper
-        set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
-    endif ()
-
     list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
-    include(sdk_versioning)
-    obtain_project_version(SDK_PROJECT_VERSION)
+
+    find_package(Git QUIET) # Adding development helper tools as git_hash built when available.
+
+    include(project_version)
+    obtain_project_version(SDK_PROJECT_VERSION aws-cpp-sdk_GIT_HASH)
 
     project("aws-cpp-sdk"
             LANGUAGES CXX
             VERSION ${SDK_PROJECT_VERSION}
             DESCRIPTION ${DESCRIPTION_STRING}
             HOMEPAGE_URL "https://docs.aws.amazon.com/sdk-for-cpp"
             )
+
+    # Setting C++ minimum requirements
     set(CMAKE_CXX_STANDARD 11)
     set(CMAKE_CXX_EXTENSIONS OFF)
     set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
     message(STATUS "Building ${PROJECT_NAME} ${CMAKE_PROJECT_VERSION}")
 
+    # Setting version header for library
+    configure_file(
```

Comment:
Producing the VersionConfig header for the build. 
Different than in LEGACY_BUILD this header is not overriding SRC files, it is produced into the  BUILD.
The SRC file version from 1.9 is marked as deprecated for future removal.